src/main/java/com/easypost/model/AddressCollection.java

```diff
@@ -1,10 +1,25 @@
 package com.easypost.model;
 
 import java.util.List;
+import java.util.Map;
+
 import lombok.Getter;
 
 @Getter
-public final class AddressCollection extends EasyPostResource {
+public final class AddressCollection extends PaginatedCollection<Address> {
     private List<Address> addresses;
-    private Boolean hasMore;
+
+    @Override
+    protected Map<String, Object> buildNextPageParameters(List<Address> addresses, Integer pageSize) {
```

Comment:
Question: have you tried to define the body of this function in the `PaginatedCollection` class?

Reply:
I'm not sure I follow.

Reply:
Instead of overriding `buildNextPageParameters` into each class and then defining the same body, have you tried to define the body in the `PaginatedCollection` so we can get rid of the override for each class?

Reply:
Ah, I see. I had that set up originally in the .NET library, but it got a bit messy having to account for types and casting. Plus. we would still need to override certain parameters anyway, but I see your point.